src/Lykke.Service.Assets.Services/WatchListService.cs

```diff
@@ -38,19 +39,41 @@ public class WatchListService : IWatchListService
         
         public async Task<IEnumerable<IWatchList>> GetAllAsync(string userId)
         {
-            var allWatchListsForUser = new Dictionary<string, IWatchList>();
-
-            foreach (var predefinedWatchList in await GetAllPredefinedAsync())
-            {
-                allWatchListsForUser[predefinedWatchList.Id] = predefinedWatchList;
-            }
-
-            foreach (var customWatchList in await GetAllCustomAsync(userId))
+            var watchLists = await Task.WhenAll
+            (
+                GetAllCustomAsync(userId),
+                GetAllPredefinedAsync()
+            );
+
+            var customWatchLists     = watchLists[0].ToArray();
+            var predefinedWatchLists = watchLists[1].ToArray();
+            var allWatchListsForUser = new List<IWatchList>
             {
-                allWatchListsForUser[customWatchList.Id] = customWatchList;
-            }
-
-            return allWatchListsForUser.Select(x => x.Value);
+                predefinedWatchLists
+                    .FirstOrDefault(IsAllAssetsWatchList)
```

Comment:
Remark: here value is selected by comparing to hardcoded string "All assets". It's better at least to use separate static resource or settings for storing strings.